packages/shared/testing/src/lib/test-factory/evm-based.erc721.test-factory.ts

```diff
@@ -0,0 +1,440 @@
+          name: '2123kd',
+          signatureId: 'cac88687-33ed-4ca1-b1fc-b02986a90975',
+          fee: {
+            gasLimit: '326452',
+            gasPrice: '20',
+          },
+        })
+
+        const json = JSON.parse(result)
+
+        expect(json.gasPrice).toBe('20000000000')
+        expectHexString(json.data)
+      })
+
+      it('valid from privateKey', async () => {
+        const result = await sdk.prepare.deploySignedTransaction({
+          chain,
+          symbol: 'ERC_SYMBOL',
+          name: 'bO6AN',
+          fromPrivateKey: testData.MAINNET.ERC_721!.PRIVATE_KEY,
+          fee: {
+            gasLimit: '326452',
+            gasPrice: '20',
+          },
+        })
+
+        expectHexString(result)
+      })
+    },
+
+    transferSignedTransaction: (
+      sdk: SdkWithErc721Functions,
+      testData: BlockchainTestData,
+      chain: 'ETH' | 'MATIC' | 'KCS' | 'ONE' | 'BSC' | 'ALGO',
+    ) => {
+      it('valid from signatureId', async () => {
+        const result = await sdk.prepare.transferSignedTransaction({
+          chain,
+          to: '0x811DfbFF13ADFBC3Cf653dCc373C03616D3471c9',
+          tokenId: '112345678',
+          contractAddress: testData.MAINNET.ERC_721!.CONTRACT_ADDRESS,
+          signatureId: 'cac88687-33ed-4ca1-b1fc-b02986a90975',
+          fee: {
+            gasLimit: '50000',
+            gasPrice: '20',
+          },
+        })
+
+        const json = JSON.parse(result)
+
+        expect(json.gasPrice).toBe('20000000000')
+        expectHexString(json.data)
+      })
+
+      it('valid from privateKey', async () => {
+        const result = await sdk.prepare.transferSignedTransaction({
+          chain,
+          to: '0x811DfbFF13ADFBC3Cf653dCc373C03616D3471c9',
+          contractAddress: testData.MAINNET.ERC_721!.CONTRACT_ADDRESS,
+          fromPrivateKey: testData.MAINNET.ERC_721!.PRIVATE_KEY,
+          tokenId: '112345678',
+          fee: {
+            gasLimit: '326452',
+            gasPrice: '20',
+          },
+        })
+
+        expectHexString(result)
+      })
+
+      it('invalid address', async () => {
+        try {
+          await sdk.prepare.transferSignedTransaction({
+            chain,
+            to: 'someinvalidaddress',
+            contractAddress: testData.MAINNET.ERC_721!.CONTRACT_ADDRESS,
+            fromPrivateKey: testData.MAINNET.ERC_721!.PRIVATE_KEY,
+            nonce: 3252345722143,
+            tokenId: '112345678',
+            fee: {
+              gasLimit: '326452',
+              gasPrice: '20',
+            },
+          })
+          fail()
+        } catch (e: any) {
+          expect(e.reason).toMatch('invalid address')
+        }
+      })
+    },
+    mintSignedTransaction: (
+      sdk: SdkWithErc721Functions,
+      testData: BlockchainTestData,
+      chain: 'ETH' | 'MATIC' | 'KCS' | 'ONE' | 'BSC',
+    ) => {
+      const provider = testData.TESTNET.ERC_721?.PROVIDER ? testData.TESTNET.ERC_721?.PROVIDER : undefined
```

Comment:
Why dont do it like that?

```suggestion
      const provider = testData.TESTNET.ERC_721?.PROVIDER
```

Also Im think if we can define provider for multiple test suites (like mint, burn, etc)?